volume-providers/vpc/vpcclient/vpcvolume/get_volume.go

```diff
@@ -45,3 +45,26 @@ func (vs *VolumeService) GetVolume(volumeID string, ctxLogger *zap.Logger) (*mod
 
 	return &volume, nil
 }
+
+// GetVolumeByName GETs /volumes
+func (vs *VolumeService) GetVolumeByName(volumeName string, ctxLogger *zap.Logger) (*models.Volume, error) {
+	ctxLogger.Debug("Entry Backend GetVolumeByName")
+	defer ctxLogger.Debug("Exit Backend GetVolumeByName")
+
+	defer util.TimeTracker("GetVolumeByName", time.Now())
+
+	// Get the volume details for a single volume, ListVolumeFilters will return only 1 volume in list
+	filters := &models.ListVolumeFilters{VolumeName: volumeName}
+	volumes, err := vs.ListVolumes(1, filters, ctxLogger)
+	if err != nil {
+		return nil, err
+	}
+
+	if volumes != nil {
+		volumeslist := volumes.Volumes
+		if volumeslist != nil && len(volumeslist) > 0 {
+			return volumeslist[0], nil
+		}
+	}
+	return nil, err
```

Comment:
We should return valid error if  `len(volumeslist) = 0`  at line no. 65

Reply:
do we return error or empty volume, checking what is return by api

Reply:
it would automatically return err if the length is zero.

We are passing the same err to provider from client.

Reply:
Right @narkarum , so no action for this point

Reply:
HA